### PR TITLE
LoadSave Tests

### PR DESCRIPTION
**Pull request by matthewtckr**
Added LoadSaveTests for a couple of steps
Added the ability to test database connection step/jobentry attributes

### CODE REVIEW

File: engine/src/org/pentaho/di/trans/steps/accessinput/AccessInputMeta.java [hunk 1]
```diff
@@ -328,6 +344,7 @@ public void setFileMask( String[] fileMask ) {
   }
 
   public void setFileRequired( String[] fileRequiredin ) {
+    this.fileRequired = new String[fileRequiredin.length];
```
Comment (mbatchelor) [inline engine/src/org/pentaho/di/trans/steps/accessinput/AccessInputMeta.java]: NPE if setFileRequired passed a null..

File: engine/src/org/pentaho/di/trans/steps/accessinput/AccessInputMeta.java [hunk 2]
```diff
@@ -349,6 +366,7 @@ public String getRequiredFilesCode( String tt ) {
   }
 
   public void setIncludeSubFolders( String[] includeSubFoldersin ) {
+    this.includeSubFolders = new String[includeSubFoldersin.length];
```
Comment (mbatchelor) [inline engine/src/org/pentaho/di/trans/steps/accessinput/AccessInputMeta.java]: NPE if setIncludeSubFolders passed a null..

File: engine/test-src/org/pentaho/di/trans/steps/loadsave/LoadSaveTester.java [hunk 3]
```diff
@@ -108,7 +112,14 @@ public FieldLoadSaveValidatorFactory getFieldLoadSaveValidatorFactory() {
...
       FieldLoadSaveValidator<?> validator = fieldLoadSaveValidatorFactory.createValidator( getter );
       try {
-        setter.set( metaToSave, validator.getTestObject() );
+        Object testValue = validator.getTestObject();
+        setter.set( metaToSave, testValue );
+        if ( validator instanceof DatabaseMetaLoadSaveValidator ) {
+          addDatabase( (DatabaseMeta) testValue );
+
+          // Steps must report used database connections, mainly for XML representation
+          Arrays.asList( metaToSave.getUsedDatabaseConnections() ).contains( testValue );
```
Comment (mbatchelor) [inline engine/test-src/org/pentaho/di/trans/steps/loadsave/LoadSaveTester.java]: This line missing an Assert ?

**Comment by lucboudreau** [comment]
Hey guys, the last wingman run flagged a new failure with this PR. Was this fixed somewhere else?

**Pull request merged**